source/app/powertabeditor.cpp

```diff
@@ -3445,8 +3462,54 @@ void PowerTabEditor::updateCommands()
         ScoreUtils::findByPosition(system.getDirections(), position) !=
         nullptr);
     myRepeatEndingCommand->setChecked(altEnding != nullptr);
+
+    // dynamics
     myDynamicCommand->setChecked(dynamic != nullptr);
 
+    if (dynamic)
+    {
+        switch (dynamic->getVolume())
+        {
+            case Dynamic::Off:
+                break;
+            case Dynamic::ppp:
+                myDynamicPPPCommand->setChecked(true);
+                break;
+            case Dynamic::pp:
+                myDynamicPPCommand->setChecked(true);
+                break;
+            case Dynamic::p:
+                myDynamicPCommand->setChecked(true);
+                break;
+            case Dynamic::mp:
+                myDynamicMPCommand->setChecked(true);
+                break;
+            case Dynamic::mf:
+                myDynamicMFCommand->setChecked(true);
+                break;
+            case Dynamic::f:
+                myDynamicFCommand->setChecked(true);
+                break;
+            case Dynamic::ff:
+                myDynamicFFCommand->setChecked(true);
+                break;
+            case Dynamic::fff:
+                myDynamicFFFCommand->setChecked(true);
+                break;
+        }
+    }
+    else
+    {
+        for (QAction *action : myDynamicGroup->actions())
```

Comment:
Can this be simplified to something like the following?
```
QAction *checked_action = myDynamicGroup->checkedAction();
if (checked_action)
    checked_action->setChecked(false);
```

Reply:
haha yeah that's much better. Done.